core/src/main/java/com/datastax/oss/driver/api/core/session/SessionBuilder.java

```diff
@@ -523,6 +525,10 @@ public SessionT build() {
           throw new IllegalStateException(
               "Can't use withCloudSecureConnectBundle and addContactPoint(s). They are mutually exclusive.");
         }
+        if (sslConfigured) {
```

Comment:
I think you should also check if SSL was configured through configuration, that is, if `datastax-java-driver.advanced.ssl-engine-factory.class` is defined.